src/components/export-gist.jsx

```diff
@@ -0,0 +1,58 @@
+/* globals XMLHttpRequest:false */
+import React from "react";
+
+export default class ExportGist extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {response: false};
+  }
+  postGist() {
+    const request = new XMLHttpRequest();
+    request.onreadystatechange = () => {
+      if (request.readyState === 4 && request.status === 201) {
+        const response = JSON.parse(request.responseText);
+        this.setState({response: response.html_url});
+      }
+    };
+    request.open("POST", "https://api.github.com/gists");
+    const data = {
+      public: true,
+      files: {
```

Comment:
what did you have in mind? Something that app developer passes through as a prop that will populate the gist when exported?